### ISSUE TITLE
Data grid / table

### ISSUE THREAD

**Issue by tony** [jonathanslenders/python-prompt-toolkit#305]
With a fix set of columns / dynamic number of rows, is there any proof of concepts out there for handling a tabular data?

**Comment by jonathanslenders** [comment]
No, sorry. Right now, there is no proof of concept of this... I keep this issue open as a reminder for when I'm going to create more examples.

**Comment by therden** [comment]
@tony:   Are you inquiring only about _displaying_ tabular data -- or about displaying an _editable_ data table -- using `prompt_toolkit`?  (Just curious...)

**Comment by tony** [comment]
displaying

**Comment by swapniljariwala** [comment]
Even I want something like this for one of my side project. I'll share the code if I complete this piece.

**Comment by jonathanslenders** [comment]
@swapniljariwala, if you want to try. Start from the prompt_toolkit 2.0 branch. (that should be easier and will be released soon.) I think you have to implement a custom FormattedTextControl with custom key bindings. The scrolling can be done by wrapping it in a Window.

**Comment by lwm** [comment]
Would https://github.com/nirum/tableprint cover your use case?

**Comment by zokrezyl** [comment]
you can use https://rich.readthedocs.io/en/stable/tables.html
1. create rich.Console with StringIO output buffer
2. print the tabel to the console
3. get the vale of the buffer (buffer.get_value())
4. render it to terminal with an ANSI object
